main.cpp

```diff
@@ -401,90 +445,189 @@ struct Engines
-    void engageThrust(Turbines turbines);
-//         2) Generate elecricity
-    void increasePower(Turbines turbines, int amountOfIncrease);
-//         3) Control speed and altitude
-    void controlSpeedAndAltitude(Turbines turbines, int amountOfChange);
+bool Engines::engageThrust(Turbines thrust, bool turbinesEngaged)
+{
+    if(turbinesEngaged == true)
+    {
+        int targetLevel = 20;
+        thrust.increaseThrustLevel(targetLevel);
+        return true;
+    }
+    else
+    {
+        return false;
+    }
+}
+void Engines::increasePower(float amountOfIncrease)
+{
+    if(amountOfIncrease == 40.f)
+    {
+        powerOutput += 40.f;
+    }
+}
+void Engines::controlSpeed()
+{
+    if(engineSpeed == 0)
+    {
+        engineSpeed += 10;
+    }
+}
 
-    Turbines engaged;
-};
 
+// UDT 7
 struct CargoSpace
 {
-    // 5 properties:
-    //     1) Volume (float)
     float volume = 64.f;
-    //     2) Weight (int)
     int weight = 1300;
-    //     3) Temperature (int)
     int temperature = 75;
-    //     4) Capacity (int)
     int capacity = 400;
-    //     5) Loading time (float)
     float loadingTime = 45;
-    // 3 things it can do:
-    //     1) Transport
+
     void transport();
-    //     2) Store
-    void store();
-    //     3) Secure
-    void secure();
+    void store(int amountOfCargo);
+    void secure(bool isDoorEnclosed);
 };
 
+void CargoSpace::transport()
+{
+    std::cout << "Cargo is being transported." << std::endl;
+}
+
+void CargoSpace::store(int amountOfCargo)
+{
+    if(amountOfCargo <= capacity)
+    {
+    std::cout << amountOfCargo << " units of cargo stored." << std::endl;
+    }
+    else
+    {
+    std::cout << "Cargo exceeds maximum capacity." << std::endl;
+    }
+}
+
+void CargoSpace::secure(bool isDoorEnclosed)
+{
+    if(isDoorEnclosed)
+    {
+    std::cout << "Cargo is secured." << std::endl;
+    }
+    else
+    {
+    std::cout << "Cargo is not secured. Please close the door." << std::endl;
+    }
+}
+
+// UDT 8
 struct Tail
 {
-    // 5 properties:
-    //     1) Tail length (int)
     int length = 55;
-    //     2) Tail width (int)
     int width = 85;
-    //     3) Elevator angle (float)
-    float elevatorAngle = 45.f;
-    //     4) Stabilizer angle (float)
-    float stabilizerAngle = 45.f;
-    //     5) Yaw control width (int)
+    int elevatorAngle = 45;
+    int stabilizerAngle = 45;
     int yawControlWidth = 30;
-    // 3 things it can do:
-    //     1) Create stability
+
     void stabilize();
-    //     2) Create lift
     void createLift();
-    //     3) Control side to side movement
     void controlSideToSide();
 };
 
+void Tail::stabilize()
+{
+    if(elevatorAngle != 0 || stabilizerAngle != 0)
+    {
+    elevatorAngle = 0;
+    stabilizerAngle = 0;
+    }
+}
+void Tail::createLift()
+{
+    if(elevatorAngle < 15)
+    {
+    elevatorAngle = 15;
+    }
+}
+void Tail::controlSideToSide()
+{
+    if(yawControlWidth < 30)
+    {
+    yawControlWidth += 10;
+    }
+}
```

Comment:
Code inside of Curly braces needs to be indented for spaces relative to those Curly braces
- [ ] revise